### PR TITLE
Refactor storage configuration

### LINKED ISSUE #296: Refactor query/builder/storage.go
Related to https://github.com/uber/jaeger/pull/235#discussion_r129092382

It should be refactored in the same way as `collector/builder/span_handler_builder.go`

### PR DESCRIPTION
**Pull request by pavolloffay**
Fixes #296

### CODE REVIEW

File: cmd/collector/app/builder/span_handler_builder.go [hunk 1]
```diff
@@ -65,69 +63,68 @@ func NewSpanHandlerBuilder(cOpts *CollectorOptions, sFlags *flags.SharedFlags, o
...
+	return spanHb, err
 }
 
-func (spanHb *SpanHandlerBuilder) initCassStore(config cascfg.SessionBuilder) func() (spanstore.Writer, error) {
-	return func() (spanstore.Writer, error) {
-		session, err := config.NewSession()
-		if err != nil {
-			return nil, err
-		}
+func (spanHb *SpanHandlerBuilder) initCassStore(config cascfg.SessionBuilder) (spanstore.Writer, error) {
```
Comment (black-adder) [inline cmd/collector/app/builder/span_handler_builder.go]: s/config/builder

File: cmd/collector/app/builder/span_handler_builder_test.go [hunk 2]
```diff
@@ -114,16 +147,34 @@ func TestNewSpanHandlerBuilderElasticSearch(t *testing.T) {
...
+	assert.NotNil(t, jaeger)
+}
+
+func TestNewSpanHandlerBuilderElasticSearchNoClient(t *testing.T) {
+	v, command := config.Viperize(AddFlags, flags.AddFlags)
+	command.ParseFlags([]string{"test", "--span-storage.type=elasticsearch"})
+	sFlags := new(flags.SharedFlags).InitFromViper(v)
+	cOpts := new(CollectorOptions).InitFromViper(v)
+
+	fmt.Println(sFlags.SpanStorage.Type)
```
Comment (black-adder) [inline cmd/collector/app/builder/span_handler_builder_test.go]: is this print required?

File: cmd/collector/app/builder/span_handler_builder.go [hunk 3]
```diff
@@ -65,69 +63,68 @@ func NewSpanHandlerBuilder(cOpts *CollectorOptions, sFlags *flags.SharedFlags, o
...
+	}
 
-		spanStore := esSpanstore.NewSpanWriter(
-			client,
-			spanHb.logger,
-			spanHb.metricsFactory,
-			esBuilder.GetNumShards(),
-			esBuilder.GetNumReplicas(),
-		)
+	spanStore := esSpanstore.NewSpanWriter(
```
Comment (black-adder) [inline cmd/collector/app/builder/span_handler_builder.go]: inline here too

File: cmd/query/app/builder/storage.go [hunk 4]
```diff
@@ -24,16 +24,21 @@ import (
 	"errors"
 	"time"
 
+	"go.uber.org/zap"
+
+	"github.com/uber/jaeger-lib/metrics"
```
Comment (black-adder) [inline cmd/query/app/builder/storage.go]: jaeger-lib should go under external github deps (middle section)

File: cmd/collector/app/builder/span_handler_builder_test.go (hunk 2 above)
Comment (pavolloffay) [inline cmd/collector/app/builder/span_handler_builder_test.go]: +1, no I forgot to remove it :/

**Comment by pavolloffay** [comment]
PR updated and rebased

**Comment by pavolloffay** [comment]
PR updated and rebased. Is there anything blocking merge?

**Pull request merged**